pkgs/games/xash3d/default.nix

```
@@ -0,0 +1,99 @@
+{ stdenv
+, lib
+, makeWrapper
+, cmake
+, pkg-config
+, SDL2
+, fontconfig
+, fetchgit
+, python3
+}:
+{ valve ? throw ''type valve directory for example nix-env --argstr valve "/home/max/valve" -iA xash3d'' } @ args:
+let
+  xash3d = fetchgit {
+    url = "https://github.com/FWGS/xash3d";
+    sha256 = "08gd84aalwb4asi1ly6rly9a9s449blqfp0qwv11akglklgd482w";
+  };
+
+  hlsdk = fetchgit {
+    url = "https://github.com/FWGS/hlsdk-xash3d";
+    sha256 = "05xz7y7qax9nkkq1b7h7llqmjqrsysw88ni5adihpc2dgr8b9wnr";
+  };
+in
+
+stdenv.mkDerivation {
+  pname = "xash3d";
+  version = "0.19.2";
+  inherit xash3d hlsdk valve;
+
+  nativeBuildInputs = [
+    makeWrapper
+    cmake
+    pkg-config
+    python3
+  ];
+
+  buildInputs = [
+    SDL2
+    fontconfig
+  ];
+
+  configurePhase =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      ''
+        cmake -DHL_SDK_DIR=$hlsdk -DXASH_SDL=yes -DXASH_VGUI=no -DXASH_64BIT=1 -S $xash3d  -B $name/xash3d
+        cmake -S $hlsdk -B $name/hlsdk -D64BIT=1
+      ''
+    else if stdenv.hostPlatform.system == "i686-linux" then
+      ''
+        cmake -DHL_SDK_DIR=$hlsdk -DXASH_SDL=yes -DXASH_VGUI=no -S $xash3d  -B $name/xash3d
+        cmake -S $hlsdk -B $name/hlsdk
+      ''
+    else
+      throw "Unsupported platform ${stdenv.hostPlatform.system}";
```

```suggestion
  configurePhase =  ''
    cmake -DHL_SDK_DIR=$hlsdk -DXASH_SDL=yes -DXASH_VGUI=no -S $xash3d  -B $name/xash3d ${lib.optionalString (stdenv.hostPlatform.system == "x86_64-linux") "-DXASH_64BIT=1"}
    cmake -S $hlsdk -B $name/hlsdk ${lib.optionalString (stdenv.hostPlatform.system == "x86_64-linux") "-D64BIT=1"}
  ''
```
Please also use cmakeFlags.